frontend/src/lib/components/QuickFilters/quickFilterFormLogic.ts

```diff
@@ -132,59 +149,47 @@ export const quickFilterFormLogic = kea<quickFilterFormLogicType>([
-            if (propertyName) {
-                actions.loadPropertySuggestions(propertyName, '')
-            }
-        },
-        loadPropertySuggestions: ({ propertyName, searchInput }) => {
-            if (propertyName) {
+    listeners(({ actions, values }) => ({
+        setQuickFilterValue: ({ name, value }) => {
+            if (name === 'propertyName' && value) {
                 actions.loadPropertyValues({
                     endpoint: undefined,
                     type: propertyFilterTypeToPropertyDefinitionType(PropertyFilterType.Event),
-                    newInput: searchInput || '',
-                    propertyKey: propertyName,
+                    newInput: '',
+                    propertyKey: value as string,
                     eventNames: [],
                     properties: [],
                 })
             }
-        },
-        submitForm: async () => {
-            if (!values.isValid) {
-                return
-            }
 
-            const payload = {
-                name: values.name.trim(),
-                property_name: values.propertyName.trim(),
-                type: 'manual-options' as const,
-                options: values.options.map((opt: QuickFilterOption) => ({
-                    id: opt.id,
-                    value: trimValue(opt.value),
-                    label: opt.label.trim(),
-                    operator: opt.operator || PropertyOperator.Exact,
-                })),
+            // Auto-fill label when value is set and label is empty
+            if (Array.isArray(name) && name[0] === 'options' && name[2] === 'value') {
+                const optionIndex = name[1] as number
+                const currentOption = values.options[optionIndex]
+                if (currentOption && !currentOption.label) {
+                    const autoLabel = Array.isArray(value) ? (value as string[]).join(', ') : ((value as string) ?? '')
+                    if (autoLabel) {
+                        actions.setQuickFilterValue(['options', optionIndex, 'label'], autoLabel)
+                    }
+                }
             }
-
-            if (props.filter) {
-                await quickFiltersLogic(props).asyncActions.updateFilter(props.filter.id, payload)
-            } else {
-                await quickFiltersLogic(props).asyncActions.createFilter(payload)
+        },
+        addOption: () => {
+            actions.setQuickFilterValue('options', [
+                ...values.options,
+                { id: uuid(), value: null, label: '', operator: PropertyOperator.Exact },
+            ])
+        },
+        removeOption: ({ index }) => {
+            actions.setQuickFilterValue(
+                'options',
+                values.options.filter((_, i) => i !== index)
+            )
+        },
+        updateOption: ({ index, updates }) => {
+            for (const [key, value] of Object.entries(updates)) {
+                actions.setQuickFilterValue(['options', index, key], value)
```

Comment:
Yeah this is strange but seems like this is kea-forms way of handling complex values